### PR TITLE
Rename 2d Nav Goal to 2d Goal Pose + move_base_simple to goal_pose

### PR DESCRIPTION
**Pull request by SteveMacenski**
Per https://github.com/ros2/rviz/issues/444

Move base isn't the used tool for ROS2 navigation, and removing `nav` from the title makes it a more general waypoint dropping tool.

### CODE REVIEW

File: rviz_common/default.rviz [hunk 1]
```diff
@@ -13,7 +13,7 @@ Panels:
   - Class: rviz_common/Tool Properties
     Expanded:
   #     - /2D Pose Estimate1
-      - /2D Nav Goal1
+      - /2D Goal Pose1
```
Comment (SteveMacenski) [inline rviz_common/default.rviz:16]: I'm not entirely clear if this is necessary, but defensively removed anything "Nav Goal" related

**Comment by SteveMacenski** [comment]
@jacobperron following up on it. I have a Nav2 PR to match https://github.com/ros-planning/navigation2/pull/1071

**Review by jacobperron — COMMENTED** [review]
cpplint is complaining about the headers guards in `goal_tool.hpp`:

```
#ifndef header guard has wrong style, please use: RVIZ_DEFAULT_PLUGINS__TOOLS__GOAL_POSE__GOAL_TOOL_HPP_
```

Otherwise, LGTM!

**Comment by SteveMacenski** [comment]
Alt Text

Will fix.

**Comment by jacobperron** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8053/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4060/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6560/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7923/)

**Comment by jacobperron** [comment]
Thanks for the contribution!

**Comment by wjwwood** [comment]
Hey guys, I just noticed this touched files in the `rviz` folder, but that's just legacy. In the future please try to avoid that unless you're migrating code out of the rviz folder into one of the new packages. Hopefully the rviz folder should eventually evaporate once everything is ported.

**Comment by SteveMacenski** [comment]
Thanks for the heads up @wjwwood. What are the big elements left to port? Is there a migration ticket?

**Comment by jacobperron** [comment]
@wjwwood Oops, my bad. I misread the file paths as starting with the name of the repo 🙃 

@SteveMacenski You can find a list of things left to port in the [README](https://github.com/ros2/rviz#not-yet-ported). There are also PRs open for a couple of them (e.g. #429 and #457).

**Comment by SteveMacenski** [comment]
@jacobperron what's the blocker on #429 ? its been sitting around for awhile

**Comment by jacobperron** [comment]
> @jacobperron what's the blocker on #429 ? its been sitting around for awhile

I'm not sure there's any blocker. We just haven't gotten around to reviewing it.
I can try and make some time.

**Comment by SteveMacenski** [comment]
Oh no pressure from me, I was just wondering if there were some actions I could help with to move it along if there was something to be done

**Pull request merged**